### DISCUSSION TITLE
LGT8F328P SD card not detected

### DISCUSSION THREAD

**Discussion by arp2** [dbuezas/lgt8fx#148] [Q&A]
I have a sketch that works fine on Arduino Nano with an OLED and an SD Reader card.
I can upload same sketch without any error on same hardware but with an LGT8F328P. OLED seems to work well but micro SD card is nor detected! 
Someone has same problem? I use CS pin10. Any help please? Thanks

**Comment by LaZsolt** [comment]
The SPI of LGT8F328P has more features than ATmega328P SPI. One of these features is 4 byte input and 4 byte output buffer. (Translated databook v1.0.5 page 195: -Input/output both have 4 buffer register.) If your SD card library not handling this buffer, SPI will not work as expected. Maybe your SD card library could use the SPI library from this lgt8f package.
But if your SD card library use hardware SPI, it won't work.

See also: https://github.com/dbuezas/lgt8fx/issues/24

**Reply by arp2** [reply]
Thanks for your reply. I use SDfat library.

**Comment by herbkonik** [comment]
Hello together,

i have the same problem as the TO :
I have a sketch wich works with an Nano clone very well, but storing to SD card dosn't. It's a simple Sketch, reading temperature, shows it at an I2C LCD an stores to the SD card.
Reading and showing at the display working fine, but i don't get the SD card to be present.
I also use the SdFat Lib, because my deek-robbot SD and RT shiesds need it ( wont work with SD Lib ).
I allready read about the probs with SdFat Lib at an other thread, but it's "a few month" old and may be someone find a solution for that...
If not, - may be one can give me some sugestions whats to do at the SdFat Lib to get it work.

Thank you in advance
Herbert

**Comment by LaZsolt** [comment]
It seems someone can use SdFat library with LGT8F328.

https://github.com/RafaelReyesCarmona/TinyTrackGPS

**Comment by herbkonik** [comment]
Samthing to read, - thank you :-)
He uses the same SdFat lib as i'm using... But i take only a quick look atm.

One more Question:
The LGT8F328P has a RTC onboard, - is there a way to keep the time if it not on power ?

**Reply by LaZsolt** [reply]
`LGT8F328P has a RTC onboard,`
Where did you get this information? I didn't know about it.

**Reply by herbkonik** [reply]
oops, my mistake... sorry
The question was for the "bluepill" and should go to stm32 forum.

The question which should come here:
I read that it's possible to use the arev pin for ADC input (A10), - is the allready implemented ?

[](https://github.com/dbuezas/lgt8fx/pull/36)

**Comment by prosper00** [comment]
For SdFat v1.x (I haven't played with the 2.x series), In SdFarConfig.h (Arduino/libraries/sdfat), set:
`#define USE_STANDARD_SPI_LIBRARY 1`

As per the comments:
```
 * If the symbol USE_STANDARD_SPI_LIBRARY is zero, an optimized custom SPI
 * driver is used if it exists.  If the symbol USE_STANDARD_SPI_LIBRARY is
 * one, the standard Arduino SPI.h library is used with SPI. If the symbol
 * USE_STANDARD_SPI_LIBRARY is two, the SPI port can be selected with the
 * constructors SdFat(SPIClass* spiPort) and SdFatEX(SPIClass* spiPort).
```

... works great then. Software SPI also works (see the rest of the comments and examples included w/SdFat)
